qcodes/instrument_drivers/tektronix/Keithley_3706A_Matrix_channels.py

```diff
@@ -0,0 +1,709 @@
+        if not self._validator(val):
+            raise InvalidValue(f'{val} is not a valid specifier. '
+                               'The specifier should be channels, channel '
+                               'ranges, slots, or "allslots".')
+        for plane in plane_specifiers:
+            if plane not in backplanes:
+                raise InvalidValue(f'{backplane} is not a valid specifier. '
+                                   'The specifier should be analog '
+                                   'backplane relay.')
+        self.write(f"channel.setbackplane('{val}', '{backplane}')")
+
+    def get_backplane(self, val: str) -> str:
+        """
+        Lists the backplane relays that are controlled in switching
+        applications for specified channels.
+
+        Args:
+            val: A string representing the channels being queried.
+        """
+        backplanes = self.get_analog_backplane_specifiers()
+        specifiers = val.split(',')
+        for element in specifiers:
+            if element in backplanes:
+                raise InvalidValue(f'{val} cannot be a analog backplane relay.')
+        if not self._validator(val):
+            raise InvalidValue(f'{val} is not a valid specifier. '
+                               'The specifier should be channels, channel '
+                               'ranges, slots, or "allslots".')
+        return self.ask(f"channel.getbackplane('{val}')")
+
+    def _get_slot_id(self) -> List[str]:
+        """
+        Returns the slot ids of the installed cards.
+        """
+        cards = self.get_switch_cards()
+        slot_id = []
+        for _, item in enumerate(cards):
+            slot_id.append('{slot_no}'.format(**item))
+        return slot_id
+
+    def _get_slot_name(self) -> List[str]:
+        """
+        Returns the names of the slots as "slotX",
+        where "X" is the slot id.
+        """
+        slot_id = self._get_slot_id()
+        slot_names = [f'slot{x}' for x in slot_id]
+        return slot_names
+
+    def _get_number_of_rows(self) -> List[int]:
+        """
+        Returns the total number of rows of the installed cards.
+        """
+        slot_id = self._get_slot_id()
+        total_number_of_rows = [int(float(self.ask(
+            f'slot[{int(i)}].rows.matrix'))) for i in slot_id]
+        return total_number_of_rows
+
+    def _get_number_of_columns(self) -> List[int]:
+        """
+        Returns the total number of columns of the installed cards.
+        """
+        slot_id = self._get_slot_id()
+        total_number_of_columns = [int(float(self.ask(
+            f'slot[{int(i)}].columns.matrix'))) for i in slot_id]
+        return total_number_of_columns
+
+    def _get_rows(self) -> List[List[str]]:
+        """
+        Returns the elements of each row.
+        """
+        total_number_of_rows = self._get_number_of_rows()
+        row_list = []
+        for _, item in enumerate(total_number_of_rows):
```

Comment:
there's no need for `enumerate` here (please also fix this in other methods)
```suggestion
        for n_rows_in_slot in total_number_of_rows:
```